pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -103,10 +101,11 @@ func (c NodeController) sync() error {
 		c.eventRecorder.Eventf("MasterNodeObserved", "Observed new master node %s", node.Name)
 		newTargetNodeStates = append(newTargetNodeStates, operatorv1.NodeStatus{NodeName: node.Name})
 	}
-	operatorStatus.NodeStatuses = newTargetNodeStates
 
-	if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
-		_, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus)
+	if _, updateError := common.UpdateStatus(c.operatorConfigClient, func(operatorStatus *operatorv1.StaticPodOperatorStatus) error {
+		operatorStatus.NodeStatuses = newTargetNodeStates
```

Comment:
Need to add a check to see if the original NodeStatuses changed.  If they did, we need to error and retry.

Reply:
> Need to add a check to see if the original NodeStatuses changed. If they did, we need to error and retry.

retry the entire loop so that we don't unnecessarily smash status results

Reply:
let's go back here to simple client UpdateStatus and use the controller work queue for retries.